pkg/smokescreen/smokescreen.go

```diff
@@ -356,7 +356,7 @@ func handleConnect(config *Config, ctx *goproxy.ProxyCtx) (*net.TCPAddr, error)
 		return nil, denyError{errors.New(decision.reason)}
 	}
 
-	resolved, err = safeResolve(config, "tcp", ctx.Req.Host)
+	resolved, decision.reason, err = safeResolve(config, "tcp", ctx.Req.Host)
```

Comment:
My assumption is that should mutate the decision object in go, but uhh this is my first time writing golang. D:

Reply:
Yeah, this should do what you expect under my understanding of golang.

Reply:
Cool, thanks.